### PR TITLE
More diagnostic info when we get a signal

### PR DESCRIPTION
**Pull request by mvidner**
This is prompted by receiving a mysterious SIGHUP in https://bugzilla.suse.com/show_bug.cgi?id=935686

### CODE REVIEW

**Comment by jreidinger** [comment]
Where is signal postmortem program? otherwise it looks good

File: liby2/src/genericfrontend.cc [hunk 1]
```diff
@@ -264,11 +294,46 @@ signal_handler (int sig)
...
+    // Note that USR1 and USR2 are handled by the logger.
+    sigaction_or_error(SIGHUP);
+    sigaction_or_error(SIGINT);
+    sigaction_or_error(SIGQUIT);
+    sigaction_or_error(SIGILL);
+    sigaction_or_error(SIGABRT);
+    sigaction_or_error(SIGFPE);
+    sigaction_or_error(SIGSEGV);
+    sigaction_or_error(SIGTERM);
+
```
Comment (aschnell) [inline liby2/src/genericfrontend.cc]: Make it DRY and avoid the define: 

```
for (int signum : { SIGHUP, SIGINT, SIGQUIT, ... })
{
    if (sigaction(signum, &sa, NULL) == -1)
        perror("sigaction " # signum);
}
```

**Comment by mvidner** [comment]
@jreidinger the postmortem program is not included, I consider it a hook to be supplied by whoever debugs a particular problem. If we come up with a hook that is useful in general, we can include it.

**Comment by mvidner** [comment]
@jreidinger so, do you think we should merge this in this state? Or what improvement should be made?

**Comment by jreidinger** [comment]
Well, to be honest it is better then before, but still it do not help with useless or confusing logs from bug reports from customers where it points to wrong location as source of exception.

**Comment by jreidinger** [comment]
so feel free to merge

**Comment by mvidner** [comment]
> where it points to wrong location as source of exception.

Do you mean the problem with showing the back-trace of a different thread than the one which caused a crash? It is a different problem, concerning internally generated signals. This PR tries to address externally generated signals.

**Comment by jreidinger** [comment]
ok, LGTM

**Pull request merged**